### ISSUE TITLE
Add support for custom DNS resolutions

### ISSUE THREAD

**Issue by dlm31** [f18m/ha-addon-dnsmasq-dhcp#19]
Having issues resolving dns names. I've declard a host named "tapo or tapo.lan", but when I try to resolve either tapo or tapo.lan I get unresolved. Trying this configuration with the original dnsmasq works, but not with this addon.

```
enable: true
port: 53
dns_domain: lan
upstream_servers:
  - 9.9.9.9
hosts:
  - host: tapo
    ip: 192.168.0.65
  - host: tapo.lan
    ip: 192.168.0.65
```

**Comment by f18m** [comment]
hey @dlm31 ,
which dnsmasq-dhcp version are you using? Is it 2.0.1 or a 1.xxx version?

can you confirm that you've trying to resolve the "tapo.lan" FQDN from a machine having its DNS server set to the IP address of the HomeAssistant instance hosting the dnsmasq-dhcp addon ? Typically you can run `resolvectl status` to verify if that's the case.

Moreover the config snippet that you provided... it does not look like a valid dnsmasq-dhcp config syntax:
* the "hosts" key does not exists
* the "dhcp_ip_address_reservations" key (the most similar one to your "hosts") should contain a YAML list with keys "mac", "name", "ip" and "link".

In other words a valid dnsmasq-dhcp config for your case would look like:

```
dns_server:
  enable: true
  port: 53
  dns_domain: lan
  upstream_servers:
    - 9.9.9.9
dhcp_ip_address_reservations:
  - name: tapo
    ip: 192.168.0.65
    mac: <your Tapo MAC address>
```

**Comment by f18m** [comment]
hey @dlm31 ,
any update you can share w.r.t. the questions in my prev comment?

**Comment by dlm31** [comment]
Hi, 

Not exactly. All those local devices i specify under the dhcp_ip_address_reservations key, do resolve properly, when I try yo resolve their ip from anaother pc on the nerwork. 

What still does not work is if I try to define oher name resolutions, or override public ones.
Example, I want the dns server to return 5.5.5.5 when asked for www.google.com.
So under the dns server settings section, some kind of key should exist to be able to define it. 

Regards,

**Comment by f18m** [comment]
> Hi,
> 
> Not exactly. All those local devices i specify under the dhcp_ip_address_reservations key, do resolve properly, when I try yo resolve their ip from anaother pc on the nerwork.
> 
> What still does not work is if I try to define oher name resolutions, or override public ones. Example, I want the dns server to return 5.5.5.5 when asked for [www.google.com](http://www.google.com). So under the dns server settings section, some kind of key should exist to be able to define it.
> 
> Regards,

Ok thanks, that's more clear now what you're asking for.
Consider that this addon is not currently exposing the full capabilities of dnsmasq... at least not yet.
I flagged this as feature request and I might be able to get there at some point in the future

**Comment by matthewturner** [comment]
No problem at all!

When I'm referring to a static ip address, I mean an ip address assigned on the device itself (not an address reservation in the DHCP server). Is this what you mean too?

Thanks again.

**Comment by f18m** [comment]
> No problem at all!
> 
> When I'm referring to a static ip address, I mean an ip address assigned on the device itself (not an address reservation in the DHCP server). Is this what you mean too?
> 
> Thanks again.

hey @matthewturner , in that case then no. 
Right now if the DHCP server does not know anything about a device (because that device does not use DHCP at all), then the associated DNS server won't be able to resolve that device (which is effectively invisible to this addon).
However this is about to change: the BETA version of this addon now supports an additional "dns_custom_hosts" list where you can provide an FQDN you like and its associated IPv4/IPv6. In this way you can give an hostname/fqdn to devices having static IPs.
